kubeflow/generate_train_eval.py

```diff
@@ -919,4 +922,1329 @@ def train_eval_sumbt(
+            s3_bucket=s3_bucket,
+            task_name=task_name,
+            s3_datadir=s3_datadir,
+            model_name_or_path=model_name_or_path,
+            input_splits=input_splits,
+            train_output_per_example=train_output_per_example,
+            nmt=nmt,
+            do_alignment=do_alignment,
+            src_lang=src_lang,
+            tgt_lang=tgt_lang,
+            dlg_side=dlg_side,
+            prepare_for_translation=False,
+            do_translation=True,
+            post_process_translation=True,
+            image=image,
+            genienlp_version=genienlp_version,
+            genie_version=genie_version,
+            thingtalk_version=thingtalk_version,
+            workdir_repo=workdir_repo,
+            workdir_version=workdir_version,
+            additional_args=additional_args
+    )
+    
+    
+@dsl.pipeline(
+    name='Postprocess for Translation',
+    description='Postprocess Translated data'
+)
+def postprocess_for_translation(
+        owner='mehrad',
+        project='spl',
+        experiment='restaurants',
+        s3_bucket='geniehai',
+        task_name='almond',
+        s3_datadir='',
+        model_name_or_path='Helsinki-NLP/opus-mt-en-{}',
+        input_splits='test+eval+train',
+        train_output_per_example='1',
+        nmt='marian',
+        do_alignment='true',
+        src_lang='en',
+        tgt_lang='',
+        dlg_side='user',
+        image='932360549041.dkr.ecr.us-west-2.amazonaws.com/genie-toolkit:latest-mehrad-spl',
+        genienlp_version='',
+        genie_version='',
+        thingtalk_version=THINGTALK_VERSION,
+        workdir_repo=GENIE_WORKDIR_REPO,
+        workdir_version=GENIE_WORKDIR_VERSION,
+        additional_args='--temperature 0.4 --repetition_penalty 1.0 --num_samples 1 --batch_size 512  --skip_heuristics --att_pooling mean --task translate --id_column 0  --input_column 1 --gold_column 1 --return_attentions --output_example_ids_too'
+):
+    all_translation_steps(
+            owner=owner,
+            project=project,
+            experiment=experiment,
+            s3_bucket=s3_bucket,
+            task_name=task_name,
+            s3_datadir=s3_datadir,
+            model_name_or_path=model_name_or_path,
+            input_splits=input_splits,
+            train_output_per_example=train_output_per_example,
+            nmt=nmt,
+            do_alignment=do_alignment,
+            src_lang=src_lang,
+            tgt_lang=tgt_lang,
+            dlg_side=dlg_side,
+            prepare_for_translation=False,
+            do_translation=False,
+            post_process_translation=True,
+            image=image,
+            genienlp_version=genienlp_version,
+            genie_version=genie_version,
+            thingtalk_version=thingtalk_version,
+            workdir_repo=workdir_repo,
+            workdir_version=workdir_version,
+            additional_args=additional_args
+    )
+    
+    
+def paraphrase_step(
```

Comment:
Is this a different paraphrase_step than the usual one? In what way?

Reply:
Yes. It's for different paraphrasing methods (masked and round-trip translation). It has different steps, contains only make commands, and sync to different s3 folders compared to generate and filter paraphrase components.